### PR TITLE
Adds Workflows

### PR DESCRIPTION
**Pull request by Shivansh-007**
Adds workflows which would:

- docker build and push to GHCR.io
- linting which would lint all pull requests and pushes to main

**MERGE #2 before this**

### CODE REVIEW

**Comment by bast0006** [comment]
> .pre-commit-config.yaml is not a file

We should probably have a pre-commit config file set up before we accept this, otherwise no idea if this works. Checks should fail though since we're not flake8 supporting.

**Comment by Shivansh-007** [comment]
> > .pre-commit-config.yaml is not a file
> 
> We should probably have a pre-commit config file set up before we accept this, otherwise no idea if this works. Checks should fail though since we're not flake8 supporting.

Already done, [here](https://github.com/onerandomusername/modmail/blob/main/.pre-commit-config.yaml), I think it is failing because this branch is not upto date with `main` and hence doesn't have that file.

**Comment by bast0006** [comment]
I'm not able to merge in those changes but you should be able to, IIRC there's a new pretty button for it too near the top (update branch or something)

**Comment by onerandomusername** [comment]
> 
> 
> I'm not able to merge in those changes but you should be able to, IIRC there's a new pretty button for it too near the top (update branch or something)

There's not a button for this unless its a fork.

**Comment by bast0006** [comment]
>        # Check out the current repository in the `sir-lancebot` subdirectory
 These also need stripping/editing/licensing

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          password: ${{ secrets.GITHUB_TOKEN  }}
+
+      # Build and push the container to the GitHub Container
+      # Repository. The container will be tagged as "latest"
+      # and with the short SHA of the commit.
+      - name: Build and push
+        uses: docker/build-push-action@v2
+        with:
+          context: modmail/
+          file: sir-lancebot/Dockerfile
```
Comment (onerandomusername) [inline .github/workflows/build.yaml:51]: This is for the wrong repo

**Review by onerandomusername — CHANGES_REQUESTED** [review]
I don't see a reason to have publishing to docker at the moment.

I think publishing to docker should be added on a later time, after a DockerFile is added.

**Review by bast0006 — APPROVED** [review]

**Review by onerandomusername — APPROVED** [review]
LGTM

**Pull request merged**